job/job.go

```diff
@@ -0,0 +1,188 @@
+package job
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type Job struct {
+	Id string `json:"id"`
+}
+
+type Status struct {
+	Status   string `json:"status"`
+	Location string `json:"location"` // Location is the unauthenticated temporary (60 secs) url used to download data
+}
+
+func CreateJob(url, authToken string, jobConfig string) (*Job, error) {
+
+	JobResponse := Job{}
+
+	payload := strings.NewReader(jobConfig)
+
+	req, _ := http.NewRequest("POST", url, payload)
+
+	req.Header.Add("authorization", fmt.Sprintf("Bearer %s", authToken))
+	req.Header.Add("content-type", "application/json")
+
+	res, err := http.DefaultClient.Do(req)
+	if res.StatusCode != http.StatusOK {
+		err = fmt.Errorf("expected %d, got %d: %s", http.StatusOK, res.StatusCode, res.Body)
+		return nil, err
+	}
+
+	defer res.Body.Close()
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	JobResponse, err = handleData(body)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("Created job %s", JobResponse.Id)
+
+	return &JobResponse, nil
+}
+
+func handleData(body []byte) (Job, error) {
+	var job = Job{}
+	err := json.Unmarshal(body, &job)
+
+	return job, err
+}
+
+func WaitForJobCompletion(url string, authToken string, job Job) (resultURL *string, err error) {
```

Comment:
again, AFAICT you could replace the `Job` argument with a simple `jobID` string argument.

Reply:
(without sacrificing clarity or anything)